system/doc/design_principles/sup_princ.xml

```diff
@@ -637,5 +655,69 @@ supervisor:terminate_child(Sup, Pid)</code>
+      <list type="ordered">
+        <item>
+          <p>The child process will have to know the pids or registered
+            names not only of the supervisor it wants to stop, but also
+            that of the supervisor's parent supervisor, in order to tell
+            it to stop the supervisor it wants to stop. This can make
+            restructuring a supervision tree difficult.</p>
+        </item>
+        <item>
+          <p><c>supervisor:terminate_child/2</c> is a blocking call that
+            will only return after the parent supervisor has finished
+            the shutdown of the supervisor that should be stopped.
+            Unless the call is made from a spawned process, this will
+            result in a deadlock, as the supervisor waits for the child
+            to exit as part of it's shutdown procedure, whereas the
+            child waits for the supervisor to shut down. If the child is
+            trapping exits, this deadlock will last until the
+            <seeguide marker="#shutdown">shutdown</seeguide>
+            timeout for the child expires.</p>
+        </item>
+        <item>
+          <p>When a supervisor is stopping a child, it will wait for the
+            shutdown to complete before accepting other calls, that is,
+            the supervisor will be unresponsive until then. If the
+            termination takes some time to complete, especially when
+            the considerations outlined in the previous point wer not
+            taken into account carefully, said supervisor might become
+            unresponsive for a long time.</p>
+        </item>
+      </list>
+      <p>Instead, it is generally a better approach to rely on 
+        <seeguide marker="#automatic-shutdown">Automatic
+        Shutdown</seeguide>.</p>
+      <list type="ordered">
+        <item>
+          <p>A child process does not need to know anything about it's
+            supervisor and it's respective parent, not even that it is
+            part of a supervision tree in the first place. It is instead
+            only the supervisor which hosts the child who must know
+            which of it's children are
+            <seeguide marker="#significant_child">significant</seeguide>
+            ones, and when to shut itself down.</p>
+        </item>
+        <item>
+          <p>A child process does not need to do anything special to
+            shut down the work unit it is part of. All it needs to do
+            is terminate normally when it concludes that it has finished
+            the task it was started for.</p>
+        </item>
+        <item>
+          <p>As a supervisor that is automatically shutting itself down
+            will perform the required shutdown steps fully independent
+            of it's parent supervisor, which will only notice that it's
+            child supervisor has terminated after it has done so, so it
+            will not be blocked.</p>
```

Comment:
The whole sentence is confusing. Maybe drop the "so"? Make it a few sentences?

Reply:
Better?

Reply:
Much.